packages/libs/src/sdk/utils/preparePaymentSplits.ts

```diff
@@ -0,0 +1,65 @@
+import { USDC } from '@audius/fixed-decimal'
+
+import { ExtendedPaymentSplit } from '../api/generated/default'
+import { LoggerService } from '../services'
+import { ClaimableTokensClient } from '../services/Solana/programs/ClaimableTokensClient'
+
+/**
+ * 1. Converts amounts to bigints
+ * 2. Spreads the extraAmount to each split
+ * 3. Creates user banks for receipients as necessary
+ * 4. Returns a simplified splits structure, a list of account/amount pairs
+ */
+export const prepareSplits = async ({
+  splits,
+  extraAmount,
+  claimableTokensClient,
+  logger
+}: {
+  splits: ExtendedPaymentSplit[]
+  extraAmount: bigint
+  claimableTokensClient: ClaimableTokensClient
+  logger: LoggerService
+}) => {
+  // Convert splits to big int and spread extra amount to every split
+  let amountSplits = splits.map((split, index, arr) => {
```

Comment:
I was thinking this too - I noted that the "extraAmount" calculation could even be part of the endpoint and the client could simple pass through the values rather than doing the claculation